File: ScottsExamples/JetpackDataBindingV2/app/src/main/java/com/ebookfrenzy/jetpackdatabindingv2/ui/main/MainFragment.java
package com.ebookfrenzy.jetpackdatabindingv2.ui.main;

//import androidx.lifecycle.ViewModelProviders;
//THE ABOVE IS DEPRECATED AND REPLACED BY THE ONE BELOW
import androidx.lifecycle.ViewModelProvider;

//LOADED WITH FRAGMENT PLUS VIEW MODEL
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

//IMPORTS ADDED HAD TO ADD THE DATABINDING TO GRADLE FILE.
import androidx.databinding.DataBindingUtil;
import com.ebookfrenzy.jetpackdatabindingv2.databinding.MainFragmentBinding;
import static com.ebookfrenzy.jetpackdatabindingv2.BR.myViewModel;

//THIS WAS ADDED BY FRAGMENT PLUS VIEW MODEL
import com.ebookfrenzy.jetpackdatabindingv2.R;

public class MainFragment extends Fragment {

    private MainViewModel mViewModel;
    public MainFragmentBinding binding;

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        //return inflater.inflate(R.layout.main_fragment, container, false);

        binding = DataBindingUtil.inflate(inflater, R.layout.main_fragment, container, false);
        binding.setLifecycleOwner(this);
        return binding.getRoot();

    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(MainViewModel.class);
        binding.setVariable(myViewModel, mViewModel);
   }

}

File: scotts_kotlin_examples/TwoWayDataBinding_eel/app/src/main/java/com/example/twowaydatabinding_eel/ui/main/MainFragment.kt
package com.example.twowaydatabinding_eel.ui.main

import androidx.lifecycle.ViewModelProvider
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import com.example.twowaydatabinding_eel.R
import com.example.twowaydatabinding_eel.databinding.FragmentMainBinding
import com.example.twowaydatabinding_eel.BR.myViewModel


class MainFragment : Fragment() {

    companion object {
        fun newInstance() = MainFragment()
    }

    private lateinit var viewModel: MainViewModel

    lateinit var binding: FragmentMainBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,savedInstanceState: Bundle?): View {
        //return inflater.inflate(R.layout.main_fragment, container, false)
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_main, container, false)
        binding.setLifecycleOwner(this)

        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        binding.setVariable(myViewModel, viewModel)

        return binding.root
    }
}